src/probs_runner/cli.py

```diff
@@ -107,13 +107,14 @@ def validate_data(obj, inputs, debug_files):
     # Load data sources
     working_dir = obj["working_dir"]
     script_source_dir = obj["script_source_dir"]
-    errors = probs_validate_data(inputs, working_dir, script_source_dir, debug_files=debug_files)
+    valid = probs_validate_data(inputs, working_dir, script_source_dir, debug_files=debug_files)
 
-    if errors == 0:
+    if valid:
         click.echo(f"Validation passed", err=True)
+        sys.exit(0)
```

Comment:
Is here a specific reason to stop (early) the program?
Usually you return a value if you want to signal something, but you keep the normal execution flow of the program

Reply:
I probably haven't phrased my question clearly.
(I am not an expert of Python, but) `sys.exit()` will immediately terminate the program, and this way of managing program flow isn't generally considered best practice in software engineering. Typically, functions return values rather than terminating the program. Even if compelling reasons to do that exist, `exit()` should only be used within the `main`.
I asked that question just because I was curious why you'd choose to terminate the program within this specific function. Also because, given the current code, executing this function always results in program termination, which seems unusual to me.

Reply:
@stefanogermano 

The code ```sys.exit()``` is in  ```cli.py``` which provides an interface for running python code (e.g. python functions) from a shell command line or from a shell script (e.g. using command ```probs-runner validate-data filename```).  When running a validation in this way the code in function ```validate_data``` https://github.com/probs-lab/probs-runner/blob/master/src/probs_runner/cli.py#L102C1-L102C45 will be the only code to run and so the ```sys.exit()``` commands will not stop any additional code from running, they are at the position where code run would normally end. They are simply there to indicate a return value to the calling script (e.g. the command shell) so that is can tell what the result of the validation is. They do not abort the calling scripts early. This is very useful in a pipeline so that if validation fails the rests of the commands in the pipeline script can be stopped from running, as used in  https://github.com/probs-lab/probs-module-integration-tests/blob/main/run-pipeline.sh in ```probs-module-integration-tests```. In this script the ```set -e``` option at the start of the script will tell the script to stop if any commands fail (i.e. return a non zero exit value).